checker.c

```diff
@@ -1,21 +1,54 @@
 #include <stdio.h>
 #include <assert.h>
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
+int alertcount=0;
+
+int checklimit(float value,float minlimit,float maxlimit)
+{
+  alertcount=0;
+  if (value < minlimit) {
+    printf("value is below the limit\n");
+    alertcount=1;
   }
-  return 1;
+  else if (value > maxlimit){
+    printf("value is above the limit\n");
+    alertcount=1;
+  }
+  else
+    printf("value is within range\n");
+ return alertcount;
+}
+
+int batteryIsOk(float temperature, float soc, float chargeRate) {
+  printf("Temperature ");
+  alertcount = checklimit(temperature,0,45);
+  printf("State of Charge ");
+  alertcount = checklimit(soc,20,80);
+  printf("Charge Rate ");
+  alertcount = checklimit(chargeRate,0,0.8);
```

Comment:
Bug: This will overwrite all previous values of `alertcount`. In effect, this function is only checking the `chargeRate`.
The tests are also testing the `chargeRate` alone.

Reply:
Thanks for pointing it out, I am reworking on it.